src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/CourseService.kt

```diff
@@ -57,8 +57,11 @@ constructor(
   }
 
   fun getCourseByOfferingId(offeringId: UUID): CourseEntity? = courseRepository.findByOfferingId(offeringId)
-  fun getAllOfferingsByOrganisationId(organisationId: String): List<OfferingEntity> =
-    offeringRepository.findAll().filter { it.organisationId == organisationId }
+
+  fun getAllOfferingsByOrganisationId(organisationId: String): List<OfferingEntity> {
+    return offeringRepository.findByOrganisationId(organisationId)
+      .filter { !it.withdrawn || referralRepository.countAllByOfferingId(it.id!!) > 0 }
```

Comment:
Can we move this call into the repository as a single sql please? rather than making 2 separate calls